app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx

```diff
@@ -89,14 +88,15 @@ export class SimpleWidget extends Component {
   };
 
   onChangeWidgetLegend = (itemId, callback = () => {}) => {
-    const uncheckedItemIndex = this.state.uncheckedLegendItems.indexOf(itemId);
-    const uncheckedLegendItems = [...this.state.uncheckedLegendItems];
+    const uncheckedItemIndex = this.uncheckedLegendItems.indexOf(itemId);
+    const uncheckedLegendItems = [...this.uncheckedLegendItems];
     if (uncheckedItemIndex !== -1) {
       uncheckedLegendItems.splice(uncheckedItemIndex, 1);
     } else {
       uncheckedLegendItems.push(itemId);
     }
-    this.setState({ uncheckedLegendItems }, callback);
+    this.uncheckedLegendItems = uncheckedLegendItems;
```

Comment:
What's the point of moving dynamic data from state to a class instance?

Reply:
This improvement is made for performance reasons.
We don't need to re-render chart components after updating the legend (it is controlled by the chart itself) see for example [this](https://github.com/reportportal/service-ui/pull/1931/files#diff-823184e033f3a12a980cda5cd468dfe9R148-R151) line